ktor-server/ktor-server-core/jvm/src/io/ktor/application/newapi/KtorFeature.kt

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.application.newapi
+
+import io.ktor.application.*
+import io.ktor.application.newapi.KtorFeature.Companion.makeFeature
+import io.ktor.request.*
+import io.ktor.response.*
+import io.ktor.routing.*
+import io.ktor.util.*
+import io.ktor.util.pipeline.*
+import io.ktor.utils.io.*
+import java.nio.channels.*
+import java.nio.channels.ByteChannel
+import kotlin.random.*
+
+public interface FeatureContext {
+    public fun onCall(callback: suspend CallExecution.(ApplicationCall) -> Unit): Unit
+    public fun onReceive(callback: suspend ReceiveExecution.(ApplicationCall) -> Unit): Unit
+    public fun onSend(callback: suspend SendExecution.(ApplicationCall) -> Unit): Unit
+}
+
+public data class Interception<T : Any>(
```

Comment:
No idea what interception is different from interceptor

Reply:
Added a javadoc.